src/librustdoc/passes/strip_priv_imports.rs

```diff
@@ -1,3 +1,8 @@
+//! Strips all private import statements (use, extern crate) from a
+//! crate. This is necessary because rustdoc will handle public imports by
+//! either inlining the item's documentation to the module or creating a
+//! "Reexports" section with the import in it. The pass ensures that all of
+//! these imports are actually relevant to documentation.
```

Comment:
Hmm, I think the simpler reason is just so the private items don't show up in the documentation - it's true that inlining has all sorts of complicated logic, but I don't think that's the main reason for stripping private imports.